static/js/apps/nl_eval/search_result.tsx

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+
+import { stringifyFn } from "../../utils/axios";
+import { BASE_URL, EmbeddingObject, MatchObject, ndcg } from "./util";
+
+interface StatVar {
+  dcid: string;
+  rank: number;
+  scores: number[];
+}
+
+interface SearchResultProps {
+  sentence: string;
+  modelName: string;
+  isExpanded: boolean;
+  goldenStatVars: string[];
+  overrideStatVars: EmbeddingObject[];
+}
+
+function dotProduct(a: number[], b: number[]): number {
+  return a.map((x, i) => a[i] * b[i]).reduce((m, n) => m + n);
```

Comment:
will `b` always have equal or greater number of elements than `a`?

Reply:
For our use case, they should always be same length vector. Return NaN if not.